data/data/powervs/dns/dns.tf

```diff
@@ -0,0 +1,52 @@
+data "ibm_dns_domain" "domain" {
+  name = var.cluster_domain
+}
+
+resource "ibm_dns_record" "bootstrap" {
+  count              = 1
+  data               = var.bootstrap_ip
+  domain_id          = data.ibm_dns_domain.domain.id
+  host               = "bootstrap.${var.cluster_id}"
+  responsible_person = "root.${var.cluster_domain}."
+  ttl                = 900
+  type               = "a"
+  tags               = [var.cluster_id, "${var.cluster_id}-bootstrap"]
+}
+resource "ibm_dns_record" "master" {
+  count              = length(var.master_ips)
+  data               = var.master_ips[count.index]
+  domain_id          = data.ibm_dns_domain.domain.id
+  host               = "master-${count.index}.${var.cluster_id}"
+  responsible_person = "root.${var.cluster_domain}."
+  ttl                = 900
+  type               = "a"
+  tags               = [var.cluster_id, "${var.cluster_id}-master"]
+}
```

Comment:
wondering why do we need DNS mappings for the bootstrap and the master nodes? I don't see these entries for other providers.

Reply:
hmm, is it because node dns resolution will be provided by mDNS ?  How are you planning to do name resolution for worker nodes ?

Reply:
yes, we might need to explore those things for the PowerVS because workers will get created on the fly post terrafrom.

Reply:
i think for workers, the internal dns would handle: https://github.com/openshift-powervs/installer/blob/add-ipi/pkg/asset/manifests/dns.go#L77
https://github.com/openshift/api/blob/master/config/v1/types_dns.go#L27-L28
also note that this is cluster-wide: https://github.com/openshift-powervs/installer/blob/add-ipi/pkg/asset/manifests/dns.go#L73

for the bootstrap, it looks like the boostrap process will fwd dns requests to the bootstrap machine: https://github.com/kubernetes-sigs/bootkube/blob/fbe1355bc63338b9ae65f8fbaa35d841e7fe9c56/cmd/render/plugin/default/asset/internal/templates.go#L860  -- so i'm assuming resolv.conf is written as part of the bootstrap ignition config?